kafka/producer/base.py

```diff
@@ -111,12 +116,20 @@ def __init__(self, client, async=False,
                  codec=None,
                  batch_send=False,
                  batch_send_every_n=BATCH_SEND_MSG_COUNT,
-                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL):
+                 batch_send_every_t=BATCH_SEND_DEFAULT_INTERVAL,
+                 maxsize=None):
 
         if batch_send:
             async = True
-            assert batch_send_every_n > 0
-            assert batch_send_every_t > 0
+            if batch_send_every_n <= 0:
+                log.exception('Batch send message count lower than zero.')
+                raise ValueError
+            if batch_send_every_t <= 0:
+                log.exception('Batch send message interval lower than zero.')
+                raise ValueError
+            if maxsize < 0:
+                log.exception('Queue size upper bound lower than zero.')
+                raise ValueError
```

Comment:
Someone might have made a comment but I would revert this back and add a additional assert for maxsize < 0.

Also, we should try to avoid adding a new exception to the class as it changes the contract.